src/Shared/FileUtilities.cs

```diff
@@ -470,16 +470,33 @@ internal static string GetFullPathNoThrow(string path)
         /// <returns></returns>
         internal static bool ComparePathsNoThrow(string first, string second, string currentDirectory)
         {
-            // todo: on xplat, figure out if file system is case sensitive or not
+            // TODO: assumption on file system case sensitivity: https://github.com/Microsoft/msbuild/issues/781
             var stringComparison = StringComparison.OrdinalIgnoreCase;
 
-            var firstFullPath = GetFullPathNoThrow(first, currentDirectory);
-            var secondFullPath = GetFullPathNoThrow(second, currentDirectory);
+            if (string.Equals(first, second, stringComparison))
+            {
+                return true;
+            }
+
+            var firstFullPath = NormalizePathForComparisonNoThrow(first, currentDirectory);
+            var secondFullPath = NormalizePathForComparisonNoThrow(second, currentDirectory);
 
             return string.Equals(firstFullPath, secondFullPath, stringComparison);
         }
 
         /// <summary>
+        /// Normalizes a path for path comparison
+        /// Does not throw IO exceptions. See <see cref="GetFullPathNoThrow(string, string)"/>
+        /// 
+        /// </summary>
+        internal static string NormalizePathForComparisonNoThrow(string path, string currentDirectory)
+        {
+            return GetFullPathNoThrow(path, currentDirectory)
+                .TrimEnd(Path.DirectorySeparatorChar)
+                .TrimEnd(Path.AltDirectorySeparatorChar);
```

Comment:
Do this in a single call (and does this one need to have the literal characters due to the forward/backslash thing?).